streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -283,6 +283,23 @@ public State state() {
         return state;
     }
 
+    /**
+     * Indicates whether or not all of the threads are fully connected
+     * 
+     * @return the connection state of the threads
+     */
+    public boolean isConnected() {
```

Comment:
I am not sure if this is the right approach for Kafka Streams. The original idea was to add a new state `DISCONNECTED` what seems to make more sense.

Note that users would need to call this method explicitly what makes it "hard to use", because it's unclear when the user might want to check this status. However, if we add a new state, users can register a callback and would get informed proactively what seems to be a better user experience.

This might imply, that we actually also want to change the consumer such that it allow us to register a callback, too.

Reply:
Compare: https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java#L74-L153

Reply:
Oh, alright. Will change it. Thanks for reviewing. :)